src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
+
+        public Node(Node<T> prev,T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexByAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> node = new Node<>(null, value, null);
+        if (index == 0) {
+            node.next = head;
+            node.prev = null;
+            head.prev = node;
+            head = node;
+        } else {
+            Node<T> currentNode = findNodeByIndex(index);
+            Node<T> newNode = currentNode.prev;
+            node.next = currentNode;
+            node.prev = newNode;
+            newNode.next = node;
+            currentNode.prev = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        T previousValue = node.item;
+        node.item = value;
+        return previousValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        unlink(node);
+        return node.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        while (node != null) {
+            if (node.item == object || node.item != null && node.item.equals(object)) {
+                unlink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(Node node) {
+        if (size == 1) {
+            size--;
+        } else if (node == head) {
+            head = node.next;
+            head.prev = null;
+            size--;
+        } else if (node == tail) {
+            tail = node.prev;
+            tail.next = null;
+            size--;
+        } else {
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+            size--;
+        }
+    }
+
+    private void checkIndex(int index) {
```

Comment:
```suggestion
    private void checkIndex(int index, int size) {
```
this way cover all cases